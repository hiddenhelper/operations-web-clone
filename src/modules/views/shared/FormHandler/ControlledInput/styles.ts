import { createStyles, makeStyles } from '@material-ui/core/styles';
import { STYLE } from '../../../../../constants';
import { toREM } from '../../../../../utils/generalUtils';

export const useStyles = makeStyles(theme =>
  createStyles({
    loginIcons: {
      '& svg': {
        fill: STYLE.COLOR.INACTIVE,
      },
    },
    loginLabels: {
      '&.MuiTypography-root': {
        fontSize: `${toREM(STYLE.FONT.SIZE.PARAGRAPH)}`,
      },

      '& .MuiTextField-root': {
        marginTop: '4px',
      },
    },
    textField: {
      '&.MuiTextField-root': {
        margin: theme.spacing(1, 0, 1, 0),
        backgroundColor: STYLE.COLOR.OCTONARY,
      },
      '&.MuiInputBase-input': {
        height: '0.85em',
        lineHeight: '27px',
        padding: '0 !important',

        '&::placeholder': {
          color: STYLE.COLOR.SECONDARY_DARKER,
        },
      },

      '&.MuiOutlinedInput-input': {
        fontSize: `${toREM(STYLE.FONT.SIZE.PARAGRAPH_BIG)}`,
        fontWeight: 500,
        position: 'relative',
        top: `${toREM(2)}`,
      },

      '&.Mui-focused': {
        '&.MuiSvgIcon-root': {
          fill: STYLE.COLOR.SECONDARY_DARKER,
        },
      },
    },
    formControl: {
      width: '100%',

      '& .MuiTypography-root': {
        fontSize: `${toREM(STYLE.FONT.SIZE.PARAGRAPH)}`,
        whiteSpace: 'nowrap',
        overflow: 'hidden',
        textOverflow: 'ellipsis',
      },

      '& .MuiOutlinedInput-root': {
        lineHeight: '27px',
        fontWeight: 600,
      },

      '& .MuiOutlinedInput-input': {
        backgroundColor: STYLE.COLOR.INPUT_BACKGROUND,
        fontWeight: 600,
      },
      '& .MuiOutlinedInput-root:hover': {
        '& .MuiOutlinedInput-notchedOutline': {
          borderColor: `${STYLE.COLOR.FOCUSED_PRIMARY}`,
          borderWidth: '2px',
        },
      },
      '&.disabled .MuiOutlinedInput-root:hover': {
        '& .MuiOutlinedInput-notchedOutline': {
          borderColor: `${STYLE.COLOR.INPUT_HOVER_DISABLED}`,
        },
      },
      '& .MuiOutlinedInput-root.Mui-focused': {
        '& .MuiOutlinedInput-notchedOutline': {
          borderColor: `${STYLE.COLOR.FOCUSED_PRIMARY}`,
          borderWidth: '2px',
        },
      },
      '& .MuiOutlinedInput-notchedOutline': {
        borderWidth: '2px',
      },

      '& .Mui-error': {
        '& .MuiOutlinedInput-notchedOutline': {
          borderColor: STYLE.COLOR.ERROR_FORM,
          borderWidth: '2px',
        },
      },
    },
    requiredMark: {
      color: STYLE.COLOR.ERROR,
    },
  })
);
