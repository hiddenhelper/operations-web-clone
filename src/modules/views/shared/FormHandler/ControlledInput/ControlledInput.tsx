import React, { memo } from 'react';

import FormControl from '@material-ui/core/FormControl';
import Typography from '@material-ui/core/Typography';

import { useStyles } from './styles';

export interface IControlledInputProps {
  label: string;
  children: React.ReactNode;
  required?: boolean;
  showMark?: boolean;
  dataTestId?: string;
  styleClass?: any;
}

const ControlledInput = ({ label, required, showMark, children, dataTestId, styleClass = '' }: IControlledInputProps) => {
  const classes = useStyles();
  const requiredMark = showMark && <span className={classes.requiredMark}>*</span>;

  return (
    <FormControl className={`${classes.formControl} ${styleClass}`}>
      <Typography color="secondary" variant="caption" align="left" component="h1">
        {label} {required && requiredMark}
      </Typography>
      <div data-testid={dataTestId}>{children}</div>
    </FormControl>
  );
};

export default memo(ControlledInput);
