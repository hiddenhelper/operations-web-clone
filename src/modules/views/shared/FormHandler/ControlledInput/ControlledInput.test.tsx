import React from 'react';
import { render, RenderResult } from '@testing-library/react';
import ControlledInput, { IControlledInputProps } from './ControlledInput';

describe('Controlled Input', () => {
  let wrapper: RenderResult;
  let props: IControlledInputProps = {
    label: 'label',
    required: true,
    showMark: true,
    children: <div>input</div>,
    dataTestId: 'testid',
  };

  it('should render with mark', () => {
    wrapper = render(<ControlledInput {...props} />);
    expect(wrapper.baseElement).toMatchSnapshot();
  });

  it('should render without mark', () => {
    props.showMark = false;
    wrapper = render(<ControlledInput {...props} />);
    expect(wrapper.baseElement).toMatchSnapshot();
  });
});
