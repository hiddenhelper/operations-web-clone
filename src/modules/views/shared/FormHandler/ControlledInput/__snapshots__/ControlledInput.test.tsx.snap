// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controlled Input should render with mark 1`] = `
<body>
  <div>
    <div
      class="MuiFormControl-root makeStyles-formControl-4 "
    >
      <h1
        class="MuiTypography-root MuiTypography-caption MuiTypography-colorSecondary MuiTypography-alignLeft"
      >
        label
         
        <span
          class="makeStyles-requiredMark-5"
        >
          *
        </span>
      </h1>
      <div
        data-testid="testid"
      >
        <div>
          input
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Controlled Input should render without mark 1`] = `
<body>
  <div>
    <div
      class="MuiFormControl-root makeStyles-formControl-9 "
    >
      <h1
        class="MuiTypography-root MuiTypography-caption MuiTypography-colorSecondary MuiTypography-alignLeft"
      >
        label
         
      </h1>
      <div
        data-testid="testid"
      >
        <div>
          input
        </div>
      </div>
    </div>
  </div>
</body>
`;
